File: implementations/react/src/patterns/structural/flyweight/helpers/make-data.ts
export function makeData(count = 2500) {
  const variants = ["pill", "rounded-outline", "chip"] as const;
  const colors = ["#10b981", "#3b82f6", "#ef4444", "#a855f7", "#111827"];
  const data: Array<{ text: string; x: number; y: number; color: string; variant: string }> = [];
  let x = 12, y = 12;

  for (let i = 0; i < count; i++) {
    const variant = variants[i % variants.length];
    const color = colors[i % colors.length];
    data.push({ text: `#${i}`, x, y, color, variant });
    x += 64;
    if (x > 960) { x = 12; y += 28; } // wrap to next "row"
  }
  return data;
}

File: implementations/flutter/design_patterns_flutter/lib/patterns/structural/flyweight/helpers/make_data.dart
import 'dart:math';

class FlyItem {
  final String variant; // pill | rounded-outline | chip
  final String text;
  final double x;
  final double y;
  final String color; // hex

  FlyItem({
    required this.variant,
    required this.text,
    required this.x,
    required this.y,
    required this.color,
  });
}

List<FlyItem> makeData(int count) {
  final rnd = Random(42);
  final variants = ['chip', 'pill', 'rounded-outline'];
  final colors = [
    '#0f172a',
    '#0ea5a4',
    '#7c3aed',
    '#dc2626',
    '#0ea5a4',
    '#334155',
  ];

  return List.generate(count, (i) {
    final v = variants[rnd.nextInt(variants.length)];
    final text = 'Item ${i + 1}';
    final x = (rnd.nextDouble() * 900).floorToDouble();
    final y = (rnd.nextDouble() * 380).floorToDouble();
    final color = colors[rnd.nextInt(colors.length)];
    return FlyItem(variant: v, text: text, x: x, y: y, color: color);
  });
}
